cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -0,0 +1,202 @@
+    pytest_selenium.send_keys("//input[@name='device_path']", '/dev/vdb')
+    pytest_selenium.click("//button[@id='save_enabled']")
+    flash_msg_loc = "//div[@id='flash_text_div']/div/strong"
+    msg = pytest_selenium.text(flash_msg_loc)
+    assert 'Attaching Cloud Volume' in msg
+    pytest_selenium.click(flash_msg_loc)
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    info_el = InfoBlock.element('Relationships', 'Cloud Volumes')
+    pytest_selenium.click(info_el)
+    params = {'Name': volume_name, 'Status': 'in-use'}
+    res = list_tbl.find_rows_by_cells(params)
+    assert res, 'Volume does not appear in instance relationships'
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_detach_volume_from_instance_page(provider):
+    """Detaches volume from instance from Instance page"""
+    vm_name = None
+    # Find an instance with attached volumes
+    for instance in provider.mgmt._get_all_instances():
+        info = instance.to_dict()
+        if info['os-extended-volumes:volumes_attached']:
+            vm_name = info['name']
+            break
+    assert vm_name, "Can't find any instance with attached volume"
+
+    Instance(vm_name, provider).load_details()
+    toolbar.select('Configuration', 'Detach a Cloud Volume from this Instance')
+    select = Select("//select[@id='volume_id']")
+    v_option = choice(select.all_options[1:])
+    select.select_by_value(v_option[1])
+    loc = "//div[@id='angular_paging_div_buttons']/button[@id='save_enabled']"
+    pytest_selenium.click(loc)
+    msg = pytest_selenium.text("//div[@id='flash_text_div']/div/strong")
+    assert 'Detaching Cloud Volume' in msg
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    info_el = InfoBlock.element('Relationships', 'Cloud Volumes')
+    pytest_selenium.click(info_el)
+    res = list_tbl.find_cell('Name', v_option[0])
+    assert not res, 'Volume does not disappear from instance relationships'
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_attach_volume_from_volume_page(provider):
+    """Attaches pre-created volume to an instance from Volume page"""
+    vms = filter(lambda vm: vm.power_state == 'ACTIVE', provider.mgmt.all_vms())
+    instance_name = choice(vms).name
+    navigate_to(Volume, 'All')
+    params = {'Status': 'available',
+              'Cloud Provider': provider.get_yaml_data()['name']}
+    list_tbl.click_row_by_cells(params, 'Name')
+    vname = pytest_selenium.text('//h1').split()[0]
+    toolbar.select('Configuration', 'Attach this Cloud Volume to an Instance')
+    select = Select("//select[@id='vm_id']")
+    assert len(select.all_options) > 0
+
+    select.select_by_value(select.get_value_by_text(instance_name))
+    pytest_selenium.send_keys("//input[@name='device_path']", '/dev/vdb')
+    pytest_selenium.click("//div[@id='buttons_on']/button")
+    msg = pytest_selenium.text("//div[@id='flash_text_div']/div/strong")
+    assert 'Attaching Cloud Volume' in msg
```

Comment:
And again, flash message handling, form input handling, form button, all of this is already defined.